src/main/java/mil/dds/anet/emails/AnetEmailAction.java

```diff
@@ -2,31 +2,19 @@
 
 import java.util.Map;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
-public abstract class AnetEmailAction {
+public interface AnetEmailAction {
 
-	protected static final int MAX_REPORT_INTENT_LENGTH = 50;
-
-	String templateName;
-	String subject;
-	
-	public abstract Map<String,Object> execute();
-	
-	public String getTemplateName() {
-		return templateName;
-	}
+	static final int MAX_REPORT_INTENT_LENGTH = 50;
 	
-	public void setTemplateName(String templateName) {
-		this.templateName = templateName;
-	}
+	void buildContext(Map<String,Object> context);
```

Comment:
(Add a space after the comma.)
I'd prefer for `buildContext` to return the context (so `Map<String, Object>` instead of `void`). This way, it supports a fluent interface, so code like:
```
		email.getAction().buildContext(context);


		return context;
```
could simply become:
```
		return email.getAction().buildContext(context);
```